pandas/tests/io/parser/test_network.py

```diff
@@ -50,151 +93,142 @@ def check_compressed_urls(salaries_table, compression, extension, mode,
-                assert not df.empty
-                true_df = local_tips.iloc[
-                    chunksize * i_chunk: chunksize * (i_chunk + 1)]
-                tm.assert_frame_equal(true_df, df)
-
-    @tm.network
-    def test_parse_public_s3_bucket_python(self):
-        for ext, comp in [('', None), ('.gz', 'gzip'), ('.bz2', 'bz2')]:
-            df = read_csv('s3://pandas-test/tips.csv' + ext, engine='python',
-                          compression=comp)
-            assert isinstance(df, DataFrame)
-            assert not df.empty
-            tm.assert_frame_equal(read_csv(
-                tm.get_data_path('tips.csv')), df)
-
-    @tm.network
-    def test_infer_s3_compression(self):
-        for ext in ['', '.gz', '.bz2']:
-            df = read_csv('s3://pandas-test/tips.csv' + ext,
-                          engine='python', compression='infer')
-            assert isinstance(df, DataFrame)
-            assert not df.empty
-            tm.assert_frame_equal(read_csv(
-                tm.get_data_path('tips.csv')), df)
-
-    @tm.network
-    def test_parse_public_s3_bucket_nrows_python(self):
-        for ext, comp in [('', None), ('.gz', 'gzip'), ('.bz2', 'bz2')]:
-            df = read_csv('s3://pandas-test/tips.csv' + ext, engine='python',
-                          nrows=10, compression=comp)
-            assert isinstance(df, DataFrame)
-            assert not df.empty
-            tm.assert_frame_equal(read_csv(
-                tm.get_data_path('tips.csv')).iloc[:10], df)
 
-    @tm.network
-    def test_s3_fails(self):
-        with pytest.raises(IOError):
-            read_csv('s3://nyqpug/asdf.csv')
+def test_s3_fails(test_s3_resource):
+    with pytest.raises(IOError):
+        read_csv('s3://nyqpug/asdf.csv')
 
-        # Receive a permission error when trying to read a private bucket.
-        # It's irrelevant here that this isn't actually a table.
-        with pytest.raises(IOError):
-            read_csv('s3://cant_get_it/')
+    # Receive a permission error when trying to read a private bucket.
+    # It's irrelevant here that this isn't actually a table.
+    with pytest.raises(IOError):
+        read_csv('s3://cant_get_it/')
 
-    @tm.network
-    def boto3_client_s3(self):
-        # see gh-16135
 
-        # boto3 is a dependency of s3fs
-        import boto3
-        client = boto3.client("s3")
+def test_read_csv__handles_boto_s3_object(test_s3_resource, tips_file):
+    # see gh-16135
 
-        key = "/tips.csv"
-        bucket = "pandas-test"
-        s3_object = client.get_object(Bucket=bucket, Key=key)
+    s3_object = test_s3_resource.meta.client.get_object(Bucket='pandas-test',
+                                                        Key='tips.csv')
 
-        result = read_csv(s3_object["Body"])
-        assert isinstance(result, DataFrame)
-        assert not result.empty
+    result = read_csv(six.BytesIO(s3_object["Body"].read()), encoding='utf8')
```

Comment:
`six` isn't a dependency of pandas (though it probably is transitively), but you can do `from pandas.compat import BytesIO`.

I'm curious why you needed it now, when we didn't before. Does simply passing `encoding='utf-8'` to `read_csv` work?

Reply:
I'm not sure why that stopped working when I starting using moto...
The error I get using `result = read_csv(s3_object["Body"], encoding='utf8')` is

```
        if not is_file_like(filepath_or_buffer):
            msg = "Invalid file path or buffer object type: {_type}"
>           raise ValueError(msg.format(_type=type(filepath_or_buffer)))
E           ValueError: Invalid file path or buffer object type: <class 'botocore.response.StreamingBody'>

pandas/io/common.py:219: ValueError
```
Making it a BytesIO explicitly makes pandas happy about it.
I'm also running python 3.6.0 locally.